src/audit-log/src/lib.rs

```diff
@@ -143,6 +143,7 @@ pub enum EventDetails {
     CreateComputeReplicaV1(CreateComputeReplicaV1),
     DropComputeReplicaV1(DropComputeReplicaV1),
     CreateSourceSinkV1(CreateSourceSinkV1),
```

Comment:
@mjibson Do you know what the migration story for these are? Are we ever expecting to deserialize old audit log messages, or can I delete old versions?

Reply:
The migration story is: this crate never migrates and must support old versions for as long as those data exist. This is important because it forces usages of this crate that read this data to make explicit choices about how to handle all possible versions. Now that I think about it, I didn't recall seeing code that handled that during builtin table updates at startup.

Reply:
Oh right it's because those are all JSON, so they just won't show up to folks who query that table. Hrm. I had always envisioned the readers of the audit log to use the crate directly out of cockroach instead of right out of our audit events table, where they then have to interpret the SQL correctly, which is not always obvious.

Reply:
Second thought about migrations: we can remove the old structs after we know there's no old data of that form anymore. We can't do that yet because we don't GC this table.

Reply:
I think the simplest thing for now is to leave the old version.